src/maven_artifact/requestor.py

```diff
@@ -1,23 +1,31 @@
 import base64
 import requests
 
+from maven_artifact.utils import Utils
+
 
 class RequestException(Exception):
     def __init__(self, msg):
         self.msg = msg
 
 
 class Requestor(object):
-    def __init__(self, username=None, password=None, user_agent="Maven Artifact Downloader/1.0"):
+    def __init__(self, username=None, password=None, token=None, user_agent="Maven Artifact Downloader/1.0"):
         self.user_agent = user_agent
         self.username = username
         self.password = password
+        self.token = token
 
     def request(self, url, onFail, onSuccess=None, method: str = "get", **kwargs):
         headers = {"User-Agent": self.user_agent}
+
         if self.username and self.password:
             token = self.username + ":" + self.password
-            headers["Authorization"] = "Basic " + base64.b64encode(token.encode()).decode()
+            headers["Authorization"] = f"Basic {base64.b64encode(token.encode()).decode()}"
+        elif Utils.is_base64(self.password):
+            headers["Authorization"] = f"Basic {self.password}"
```

Comment:
perhaps we should allow users to specify the Authorization header directly here.
Then we dont have to care about what users actually auths with.

Reply:
I don't think that there is a need. This works with basic auth and OAuth. If there is a need then we can add it.
But, I rather not add things that might be used.